src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -259,7 +260,10 @@ public void Clear()
             int count = _count;
             if (count > 0)
             {
-                Array.Clear(_buckets, 0, _buckets.Length);
+                Debug.Assert(_buckets != null, "_buckets should be non-null");
+                Debug.Assert(_entries != null, "_entries should be non-null");
+
+                Array.Clear(_buckets, 0, _buckets!.Length);
```

Comment:
Why is a `!` needed here given the assert just above?

Reply:
Ha I think I added the Debug.Assert after adding the `!` and forgot to remove `!`

Reply:
Will you be removing the `!`?

Reply:
Yes.